src/dlc-handlers/dfns-dlc-handler.ts

```diff
@@ -0,0 +1,369 @@
+
+  constructor(
+    dfnsAppID: string,
+    appBaseURL: string,
+    dfnsAuthToken: string,
+    bitcoinNetwork: Network,
+    bitcoinBlockchainAPI: string,
+    bitcoinBlockchainFeeRecommendationAPI: string
+  ) {
+    this.dfnsDelegatedAPIClient = new DfnsDelegatedApiClient({
+      baseUrl: appBaseURL,
+      appId: dfnsAppID,
+      authToken: dfnsAuthToken,
+    });
+    this.bitcoinBlockchainAPI = bitcoinBlockchainAPI;
+    this.bitcoinBlockchainFeeRecommendationAPI = bitcoinBlockchainFeeRecommendationAPI;
+    this.bitcoinNetwork = bitcoinNetwork;
+  }
+
+  private setPayment(fundingPayment: P2TROut, multisigPayment: P2TROut): void {
+    this.payment = {
+      fundingPayment,
+      multisigPayment,
+    };
+  }
+
+  private getPayment(): PaymentInformation {
+    if (!this.payment) {
+      throw new Error('Payment Information not set');
+    }
+    return this.payment;
+  }
+
+  async getWallets(): Promise<ListWalletsResponse> {
+    try {
+      return await this.dfnsDelegatedAPIClient.wallets.listWallets();
+    } catch (error: any) {
+      throw new Error(`Error fetching wallets: ${error}`);
+    }
+  }
+
+  async initializeWalletByID(dfnsWalletID: string): Promise<void> {
+    try {
+      const dfnsWallet = await this.dfnsDelegatedAPIClient.wallets.getWallet({
+        walletId: dfnsWalletID,
+      });
+      this.setDFNSWalletID(dfnsWallet.id);
+      this.setTaprootDerivedPublicKey(dfnsWallet.signingKey.publicKey);
+    } catch (error: any) {
+      throw new Error(`Error fetching wallet: ${error}`);
+    }
+  }
+
+  setDFNSWalletID(dfnsWalletID: string): void {
+    this.dfnsWalletID = dfnsWalletID;
+  }
+
+  getDFNSWalletID(): string {
+    if (!this.dfnsWalletID) {
+      throw new Error('DFNS Wallet ID not set');
+    }
+    return this.dfnsWalletID;
+  }
+
+  setTaprootDerivedPublicKey(taprootDerivedPublicKey: string): void {
+    this.taprootDerivedPublicKey = taprootDerivedPublicKey;
+  }
+
+  getTaprootDerivedPublicKey(): string {
+    if (!this.taprootDerivedPublicKey) {
+      throw new Error('Taproot Derived Public Key not set');
+    }
+    return this.taprootDerivedPublicKey;
+  }
+
+  getTaprootTweakedPublicKey(): string {
+    if (!this.payment) {
+      throw new Error('Payment Information not set');
+    }
+    return bytesToHex((this.payment.fundingPayment as P2TROut).tweakedPubkey);
+  }
+
+  getVaultRelatedAddress(paymentType: 'funding' | 'multisig'): string {
+    const payment = this.getPayment();
+
+    if (payment === undefined) {
+      throw new Error('Payment Objects have not been set');
+    }
+
+    switch (paymentType) {
+      case 'funding':
+        if (!payment.fundingPayment.address) {
+          throw new Error('Funding Payment Address is undefined');
+        }
+        return payment.fundingPayment.address;
+      case 'multisig':
+        if (!payment.multisigPayment.address) {
+          throw new Error('Taproot Multisig Payment Address is undefined');
```

Comment:
like these kinds of error messages must have lots of duplication between the different handlers